### DISCUSSION TITLE
Alte Entitäten im Energy Dashboard - bisherige Statistiken übernehmen

### DISCUSSION THREAD

**Discussion by phrozen77** [SukramJ/aiohomematic#1283] [General]
Hallo zusammen,

eben habe ich in den sauren Apfel gebissen, die alte Konfigurations-gebundene Integration deaktiviert und die neue in HACS installiert und konfiguriert.

Die Automationen in HA anzupassen war weitestgehend einfach - aber wie erwartet etwas "Arbeit zu Fuß".

Soweit so gut - ich habe allerdings einen Zentner HM-ES-PMSw1-Pl (also Schaltaktoren mit Leistungsmessung) die ich auch im Energy Dashboard überwache.

Alte Beispiel-Entität:

sensor.peq0413666_energy_counter

... heißt jetzt sensor.gerätename_energiezahler

Somit - Entität nicht gefunden, bisherige Statistiken wertlos - und das sind leider schon ein paar Jahre an historischen Daten.

Habt ihr einen (schönen|einfachen|überhaupt einen) Migrationspfad für diesen Fall?

Danke :)

**Comment by SukramJ** [comment]
> Habt ihr einen (schönen|einfachen|überhaupt einen) Migrationspfad für diesen Fall?

Ich hab keinen Migrationspfad. Hast du mal die neue Entität umbenannt?

**Reply by phrozen77** [reply]
Die neue Entität in die alte? Nein, die Richtung noch nicht - andersrum soll es angeblich gehen, tut es aber nicht (Fehlermeldung: Entity with this ID is already registered).

Klar, die Entität mit dem neuen Namen existiert natürlich - soweit so korrekt...

Und: die neuen Entitäten jetzt in die alten Namen umbenennen - ich weiß nicht was die/eure Integration dazu sagt, wenn sie auf einmal Entitäten die da sein sollten, nicht mehr gefunden werden?

Ganz davon abgesehen: mir gefallen die neuen Entitäts"klar"namen durchaus 👍

**Reply by SukramJ** [reply]
Ohne die alte Integration zu installieren wird das wohl nichts, sonst ist die entity_id nicht frei, und kann nicht wieder verwendet werden.
Verwendete entity_ids können zwar umbenannt werden, aber die gehören dann weiterhin zur alten Integration.

Vielleicht findest du ja noch was raus.

**Reply by Lemocin** [reply]
Been there, done that. Alte Entitäts ID merken und löschen oder umbenennen, neue Entitäts ID mit der ID der alten ersetzen. So hat das bei mir schon einige Male funktioniert.

**Comment by danielperna84** [comment]
Keine Lösung für dein Problem, aber ich mach's so: alles, was ich _ewig_ behalten will, kommt bei mir in InfluxDB. Und dann habe ich dort z.B. "Gesamtverbrauch alt" (mit Daten von einer alten Entity) und "Gesamtverbrauch" (wo die Daten der neuen Entity reinlaufen). Es kann schließlich immer wieder mal vorkommen, dass Geräte ausgetauscht werden. Und mir ist's zu blöd mich da mit HA rum zu ärgern, wenn man das mit InfluxDB + Grafana ganz einfach lösen kann.
